### ISSUE TITLE
multiply shuttle and jogwheel

### ISSUE THREAD

**Issue by notgernot** [abourget/shuttle-go#9]
Hi, is it possible to multiply jogwheel or shuttle

like M1+JogL or M1+S1
cause if i try it like that it does actually not multiply functionality.
And could anyone of you help me with the shuttle-go.json
i would like to implement a kind of if -> then function
 in that kind
if ->any windwoclass name like Lightworks, kdlive,......fits use the following button-functions
if none of them fits -> use the default functions like jogwheel-systemaudio, F1 systemMute, media-Play,....

thanks alot notger

PS.: this is a great think thanks for bringing the shuttlepro to linuxworld

**Comment by abourget** [comment]
Yeah right now the Jog and Shuttle are sort of specific, they can't be _modified_ by some other buttons.  That shouldn't be tough to implement, but I won't have time to do it right now, though.

Regarding the window title, what is specified in `matchin_window_titles` is a list of regular expressions, so you can do things like `Lightworks|kdenlive`.. that's a logical OR between the two strings.. you can specifiy multiple strings, and different sections in `apps` .. do you see some combinations that are missing?  Is it that you would like to default back to what the Shuttle Pro does when `shuttle-go` isn't running?  I guess that would mean having a map of what it does _naturally_ as detected by X11, and draft a config for it.

**Comment by notgernot** [comment]
Hi, i guess i have to be honest that i just have quite basic programming skills.
my c++ time is quite a while ago, as php or others.
so i see that .json is a whole new world, that i actually do not really want to  understand fully.
So no i do not really want to go back to the rural functionality of the "just plugged in shuttle"

my only aim would be something like

{
    "apps": [
        {
            "name": "Lightworks",
            "match_window_titles": [
                "^Lightworks$"
.............
},
{
            "name": "Ardour",
            "match_window_titles": [
                "Ardour$"
.................................
},
{
            "name": "anything_else",
            "match_window_titles": [
                "anything/anywhere that was not specified before..fexpl. the cursor is on the empty Desktop, or using Thunderbird,Firefox,....."
.........................
                "B3+F2": "XF86MonBrightnessUp",
                "B3+F1": "XF86MonBrightnessDown",
                "F7": "XF86AudioPlay",
                "F9": "XF86AudioMute",
                "JogL": "Control+minus",
                "JogR": "Control+plus",
                "SlowJogL": "XF86AudioRaiseVolume //lauter",
                "SlowJogR": "XF86AudioLowerVolume //leiser",
......................

So having a "default" setup that is always valid,.. not based on any window_titles matching
........ with basic functions witch are nice to have and use in most programms or situations like scroll up down, raise/lower systemvolume, mute, play, brightness..............
and as soon as the window_title-is_matching lets say "Ardour$"-> the shuttle should be in Ardour-mode again.

hope that makes my aim more clear
...Im coming from SHUTTLEPRO from https://github.com/agraef/ShuttlePRO
it has a "default" setup that is always working as soon as you start shuttlepro... besides specific window_title based button-setups.......but sadly it does not allow  modifying shuttlebuttons
